win32/devdoc/gballoc_ll_jemalloc.md

```diff
@@ -165,4 +167,28 @@ static void jemalloc_print_stats_callback(void* context, const char* text)
 
 **SRS_GBALLOC_LL_JEMALLOC_01_009: [** If `text` is NULL, `jemalloc_print_stats_callback` shall return. **]**
 
-**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
\ No newline at end of file
+**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
+
+### gballoc_ll_set_decay
+
+```c
+MOCKABLE_FUNCTION(, int, gballoc_ll_set_decay, int64_t, decay_milliseconds);
+```
+
+`gballoc_ll_set_decay` sets the dirty and muzzy pages decay time to `decay_milliseconds` milliseconds.
+
+**SRS_GBALLOC_LL_JEMALLOC_28_001: [** `gballoc_ll_set_decay` shall advance jemalloc's internal epoch counter. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_002: [** `gballoc_ll_set_decay` shall set the dirty decay time for new arenas to `decay_milliseconds` milliseconds. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_003: [** `gballoc_ll_set_decay` shall set the muzzy decay time for new arenas to `decay_milliseconds` milliseconds. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_004: [** `gballoc_ll_set_decay` shall fetch the number of existing jemalloc arenas. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_005: [** For each existing arena: **]**
+
+- **SRS_GBALLOC_LL_JEMALLOC_28_006: [** `gballoc_ll_set_decay` shall set the dirty decay time for the arena to `decay_milliseconds` milliseconds. **]**
+
+- **SRS_GBALLOC_LL_JEMALLOC_28_007: [** `gballoc_ll_set_decay` shall set the muzzy decay time for the arena to `decay_milliseconds` milliseconds. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_008: [** If there are any errors, `gballoc_ll_set_decay` shall fail and return a non-zero value. **]**
```

Comment:
>If there are any errors [](http://example.com/codeflow?start=38&length=23)

This is a bit weird here, seeing as there's a "for each" in line 188 of this .md

What happens if there are 3 arenas... and... the 2nd one fails?

Will the third arena have its dirty/muzzy decay timer adjusted? Will it be skipped? Seeing as we'd already know the return value from the function ("error")

Reply:
we are now doing a best effort undo.